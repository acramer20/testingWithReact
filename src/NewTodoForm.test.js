import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function() {
    render(<NewTodoForm/>);
});
it("Matches the snapshot", function() {
    const {asFragment} = render(<NewTodoForm/>);
    expect(asFragment()).toMatchSnapshot();
})
it("creates new todo on form submit", function() {
    const createFakeTodo = jest.fn();
    const {getByText} = render(<NewTodoForm createTodo={createFakeTodo}/>);
    const createButton = getByText("Add new todo!");
    fireEvent.click(createButton);
    expect(createFakeTodo).toHaveBeenCalled();
})