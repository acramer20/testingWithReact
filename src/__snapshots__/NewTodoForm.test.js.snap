// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add new todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
